samples/reducer.js

```diff
@@ -54,9 +68,47 @@ function createReducer (config) {
     }),
     [loadSampleEnd]: (state, action) => ({
       ...state, isLoading: false
+    }),
+    [createSample]: (state, action) => {
+      const file = action.payload
+      assert(file && file.path, 'Cannot createSample without file && file.path')
+
+      return loop({
+        ...state, isCreating: true
+      }, Effects.batch([
+        Effects.promise(runCreateSample, file),
+        Effects.constant(createSampleEnd())
```

Comment:
note i tried the alternative approach in [the next pr](https://github.com/linxmix/linx-electron/pull/11/files#diff-def1f84f5412306bde74c0bb223f28d7R49), copying the file first then just calling `readSample`